pmd-java/src/main/java/net/sourceforge/pmd/lang/java/oom/Metrics.java

```diff
@@ -101,7 +102,7 @@ public static double get(OperationMetricKey key, ASTMethodOrConstructorDeclarati
      *
      * @return The value of the metric, or {@code Double.NaN} if the value couln't be computed
      */
-    public static double get(OperationMetricKey key, ASTMethodOrConstructorDeclaration node, MetricVersion version) {
+    public static double get(MetricKey<OperationMetric> key, ASTMethodOrConstructorDeclaration node, MetricVersion version) {
```

Comment:
I tried finding more details on the `MetricVersion` digging through /master here on github, but didn't any classes.  Not in IDE right now, so discovery is merely manual.  Are there usages?  The Metric enums didn't seem use this per the comments on the interface.  _Anyway_, If the metric key and version are related, I'm curious why it makes sense to _separate_ them in argument order for some of these overloads.  For example here, the argument order is `Key, Node, Version`.

I have other questions, but I think I need to understand the relationships in this object model better before asking.  I'll find some time to do so.  Feel free to point me toward any resources I could use to enlighten myself.